### PR TITLE
init cli improvements: check the current version of RNmacOS, other display tweaks

### PR DESCRIPTION
**Pull request by Simek**
#### Please select one of the following
- [x] I am making a fix / change for the macOS implementation of react-native

## Summary

This PR introduces the following changes to the `react-native-macos-init` script:
* check the currently installed version of `react-native-macos`, skip installation if it is latest available version
* refactor `getReactNativeVersion` to the `getPackageVersion` (more generalised)
* utilize `verbose` flag in `react-native-macos` installation exec
* store package names in the constants
* update and simplify display aspect of the script (`printPkg` method for pretty-printing packages and other tweaks), update colors (this can be easily changed now): 
  * package name changed from the green to the yellow
  * `prerelease` warning uses yellow background instead yellow text
* `printError` function which prints decorated error messages (red + bold)
* punctuation marks in CLI messages, ellipsis instead of three dots (my code OCD 😉)

Since the changes above has not been request (and some of them apply only for those who are upgrading) I can understand if there will be a lot of review feedback and request for the changes but I'm happy to work further on those improvements and bring this PR to the more meaningful state. 

In the future PRs I want to adjust the display and processes inside `local-cli` too, for example changed file prompts (leave or replace). 

In a long term it will be nice if `react-native-macos-init` can also handle or at least help with the upgrading existing  project which already is using the RN macOS to the latest version.

## Changelog

[Internal] [Added] - [init] Check the currently installed version of `react-native-macos`, skip installation if it is latest
[Internal] [Changed] - [init] Utilize `verbose` flag in `react-native-macos` installation exec

## Test Plan

I have tested those changes locally by linking working copy of the `react-native-macos-init` package in the test macOS app.

## Preview

Successful init:
Screenshot 2020-05-17 at 17 03 38

Successful init (skip installation):
Screenshot 2020-05-17 at 17 06 44

Decorated errors:
Screenshot 2020-05-17 at 16 40 34
Screenshot 2020-05-17 at 16 50 08

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-macos/pull/370)

### CODE REVIEW

**Comment by Simek** [comment]
CI fail looks like not related to this PR:
```
Failing tests:
	RNTester-macOSIntegrationTests:
		-[RNTesterLoadAllPages JSResponderHandlerExample]

** TEST FAILED **

##[error]Error: /usr/bin/xcodebuild failed with return code: 65
```

File: packages/react-native-macos-init/src/cli.ts [hunk 1]
```diff
@@ -275,29 +300,29 @@ You can either downgrade your version of ${chalk.green(
...
-    execSync(`${pkgmgr} "react-native-macos@${version}"`, execOptions);
-    console.log(
-      chalk.green(`react-native-macos@${version} successfully installed.`),
-    );
+    if (reactNativeMacOSResolvedVersion !== reactNativeMacOSVersion) {
+      console.log(`${reactNativeMacOSVersion ? 'Upgrading to' : 'Installing'} ${pkgLatest}…`);
+
+      const pkgmgr = isProjectUsingYarn(process.cwd())
+        ? `yarn add${verbose ? '' : ' -s'}`
+        : `npm install --save${verbose ? '' : ' --silent'}`;
```
Comment (alloy) [inline packages/react-native-macos-init/src/cli.ts:310]: Nice 👌

File: packages/react-native-macos-init/src/cli.ts [hunk 2]
```diff
@@ -47,56 +47,75 @@ const EXITCODE_NO_REACTNATIVE_FOUND = 5;
...
 function reactNativeMacOSGeneratePath() {
-  return require.resolve('react-native-macos/local-cli/generate-macos.js', {
+  return require.resolve(`${MACOSPKG}/local-cli/generate-macos.js`, {
     paths: [process.cwd()],
   });
 }
 
 function getReactNativeAppName() {
-  console.log('Reading application name from package.json...');
+  console.log(`Reading ${chalk.cyan('application name')} from package.json…`);
```
[the remark this answers is not in the document]
Comment (tom-un) [reply packages/react-native-macos-init/src/cli.ts:60]: The messages are parallel to what react-native-windows currently emits.   I don't have a strong opinion on the verbosity level except that the two tools look and feel consistent.

**Comment by Simek** [comment]
> The current failure in the CI just means you need to run `yarn change` from `packages/react-native-macos-init` and fill out the prompts. It will write a change log file that will trigger beachball to publish a new version of the package to npmjs.org.

@tom-un I have pushed the generated file, hope I have done this right.

**Comment by Simek** [comment]
One thing that I'm not sure is if `execOptions` ( and `stdio`) can be removed since flags are responsible for handling verbose? Only two minor changes for me testing this were:
* temporary visible installation progress (Yarn)
* `warning` word is yellow at the beginning of the `peerDependency` warning (see screenshot above) (Yarn).

Screenshot 2020-05-18 at 19 51 54

**Comment by tom-un** [comment]
There was an issue in the PR and publish YAML scripts when invoking beachball.   It was fixed with https://github.com/microsoft/react-native-macos/pull/374.   Please merge master into your branch and the CI issue should be resolved.

**Pull request merged**